lib/utils/job-utils/racadm-tool.js

```diff
@@ -192,29 +220,136 @@ function racadmFactory(
      * @param {string} host
      * @param {string} user
      * @param {string} password
-     * @param {Object} cifsConfig -  includes samba password, username and bios.xml file path
+     * @param {Object} cifsConfig -  includes samba password, username and image file path
+     * Password and username are only required when fiel path is a cifs share folder.
      */
-    RacadmTool.prototype.updateIdracImage = function(host, user, password, cifsConfig) {
-        var command = '';
-        var cifsUser = cifsConfig.user,
-            cifsPassword = cifsConfig.password;
-        assert.string(cifsUser);
-        assert.string(cifsPassword);
+    RacadmTool.prototype.updateFirmware = function(host, user, password, cifsConfig) {
+        var command = '', cifsUser, cifsPassword, self = this;
+        cifsConfig = cifsConfig || {};
+        if (!cifsConfig.filePath){
+            throw new Error('Can not find file path required for iDRAC image update');
```

Comment:
This error should be wrapped as a rejected Promise, so that the external caller can expect all error is returned from Promise and will avoid the `try/catch` to handle this specified error.